src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -153,10 +150,16 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
             }
         }
 
-        async void Consumer_Received(object sender, BasicDeliverEventArgs eventArgs)
+        async Task Consumer_Received(object sender, BasicDeliverEventArgs eventArgs)
         {
-            var eventRaisingThreadId = Thread.CurrentThread.ManagedThreadId;
+            if (messageProcessing.Token.IsCancellationRequested)
+            {
+                return;
+            }
+
+            Interlocked.Increment(ref numberOfExecutingReceives);
 
+            // technically we don't need this anymore
```

Comment:
Should I remove this one? Technically with the new concurrency code and the optimized buffer handling we don't require this. It is only here because core doesn't support readonly memory

Reply:
Yeah I think it makes sense to move this back into the `Process` method and remove `eventArgsCopy` since it isn't needed now.